syntax = "proto3";

package practice.rpc;

import "nanopb.proto";

message WifiSettings {
  string ssid = 1 [(nanopb).max_size = 32];
  string password = 2 [(nanopb).max_size = 64];
}

message LedRequest {
  bool on = 1;
}

message LedResponse {}

message EchoRequest {
  string msg = 1 [(nanopb).max_size = 128];
}

message EchoResponse {
  string msg = 1 [(nanopb).max_size = 128];
}

message SensorRequest {}

message SensorResponse {
  float temperature = 1;
  float humidity = 2;
}

message Empty {}

service DeviceService {
  rpc SetLed(LedRequest) returns (LedResponse);
  rpc Echo(EchoRequest) returns (EchoResponse);
  rpc GetSensorData(SensorRequest) returns (SensorResponse);
  rpc StartSensorStream(SensorRequest) returns (stream SensorResponse);
  rpc StopSensorStream(Empty) returns (Empty);
  rpc ConfigureWifi(WifiSettings) returns (Empty);
}
